docs/orchestrating-elastic-stack-applications/elasticsearch/virtual-memory.asciidoc

```diff
@@ -37,6 +41,69 @@ EOF
 
 Note that this requires the ability to run privileged containers, which is likely not the case on many secure clusters.
 
+== Using a Daemonset to set virtual memory
+
+To use a Daemonset that changes the host kernel setting on all nodes in your cluster and run an Elasticsearch instance that waits for the kernel setting to be in place:
+
+[source,yaml,subs="attributes,+macros"]
+----
+cat $$<<$$EOF | kubectl apply -n elastic -f -
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: max-map-count-setter
+  labels:
+    k8s-app: max-map-count-setter
+spec:
+  selector:
+    matchLabels:
+      name: max-map-count-setter
+  template:
+    metadata:
+      labels:
+        name: max-map-count-setter
+    spec:
+      initContainers:
+        - name: max-map-count-setter
+          image: docker.io/bash:5.2.15
+          resources:
+            limits:
+              cpu: 100m
+              memory: 32Mi
+          securityContext:
+            privileged: true
+            runAsUser: 0
+          command: ['/usr/local/bin/bash', '-e', '-c', 'echo 262144 > /proc/sys/vm/max_map_count']
+      containers:
+        - name: sleep
+          image: docker.io/bash:5.2.15
+          command: ['sleep', 'infinity']
+---
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
```

Comment:
nit: I think I would split this manifest into 2 parts to make it more obvious what are their purposes.